src/atlas/array/native/NativeArray.cc

```diff
@@ -62,7 +62,7 @@ Array* Array::wrap(Value* data, const ArrayShape& shape) {
 template <typename Value>
 Array* Array::wrap(Value* data, const ArraySpec& spec) {
     size_t size = spec.size();
-    return new ArrayT<Value>(new native::WrappedDataStore<Value>(data, size), spec);
+    return new ArrayT<Value>(new native::WrappedDataStore<Value>(data, spec, size), spec);
```

Comment:
So we can remove the extraction of `size` in this case, and only pass `spec` to WrappedDataStore

Reply:
Thanks for the comment. I have taken your suggestion directly in 6811aa37f.